### PR TITLE
Update RealConnection.java

### PR DESCRIPTION
**Pull request by DevResRod**
Re-create the connection when Proxy server closes the connection after an unauthorized request, to avoid IO issue with closed socket.

### CODE REVIEW

**Comment by swankjesse** [comment]
Please don’t combine behavior changes with refactorings. Ie. don’t extract methods.

Please include a test case that fails without the behavior change.

**Comment by DevResRod** [comment]
I have reverted the method extraction (refactor) to ease the change tracking. Is it enought?
Also, added a new test method in CallTest.java that fails with the original code and succeeds with this commit.

File: okhttp/src/main/java/com/squareup/okhttp/internal/io/RealConnection.java [hunk 1]
```diff
@@ -292,7 +293,16 @@ private void createTunnel(int readTimeout, int writeTimeout) throws IOException
         case HTTP_PROXY_AUTH:
           tunnelRequest = OkHeaders.processAuthHeader(
               route.getAddress().getAuthenticator(), response, route.getProxy());
-          if (tunnelRequest != null) continue;
+          if (tunnelRequest != null) {
+              String header = response.header("Connection");
+              if (header != null && header.toLowerCase().contains("close")) {
```
Comment (yschimke) [inline okhttp/src/main/java/com/squareup/okhttp/internal/io/RealConnection.java]: .contains() seems a bit weak here, value is generally keep-alive or close.

**Comment by yschimke** [comment]
Any plans to migrate this to kotlin?  Seems like a useful fix, but not essential (it should recover anyway) unless you suffer this daily with additional latency and timeouts.

**Pull request closed without merging**